woocommerce/woocommerce-gutenberg-products-block
# Subscriptions Shipping Packages Integration Points

This PR creates a new `__experimentalOrderShippingPackages ` slot fill for shipping packages, and adds several integration points for subscriptions to add it's data ([the integration on the subscriptions side, including new endpoint data, can be seen in this PR](https://github.com/woocommerce/woocommerce-subscriptions/pull/3959)).

The initial package (renamed via a server side hook) is added via the SlotFill. The collapsible state of shipping methods is based on whether it's enabled (it is on cart, and isn't on checkout) and if there are more than 1 shipping packages, or more than 1 SlotFills present.

The client side subscriptions integration is also in this PR (within the cart checkout components code) and will need to be moved once components it requires are exported. Right now it's using `useStoreCart` and `Packages` from the `ShippingRateControl` package.

For the API endpoint integration, we needed to expose the schemas. Subscriptions is using this to create it's own array of shipping packages using the core schema.

This PR was created as an alternative to https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3672.

Closes #3241

**To test**
- You must be running subscriptions, and have this branch active and built https://github.com/woocommerce/woocommerce-subscriptions/pull/3959
- Create at least 2 subscription products with differing schedules, e.g. a monthly and a weekly subscription
- Add both to your cart
- See the shipping rates in the cart totals sidebar. They should be collapsible. 
- Go to checkout. You should see subscription shipping rates below the regular rates. They will not be collapsible
